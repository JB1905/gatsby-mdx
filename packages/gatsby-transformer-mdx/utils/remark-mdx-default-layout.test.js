const mdx = require('@mdx-js/mdx');
const plugin = require('./remark-mdx-default-layout');

const MDX_NO_LAYOUT_PLAIN = `
# Hello world

- item 1
- item 2
`;

const MDX_NO_LAYOUT_WITH_IMPORTS = `
import Layout from "../components";

# Hello world

- item 1
- item 2

<Layout title="Yep" />
`;

const MDX_WITH_DEFINED_LAYOUT = `
export default ({children}) => <div style={{border: '1px solid red'}}>{children}</div>

# The Title

Lore ipsum
`;

const MDX_WITH_IMPORTED_LAYOUT = `
import Layout from "somewhere";

export default Layout

## Help

> node fix-stuff.js
`;

it('should not override defined layout', async () => {
  const result0 = await mdx(MDX_WITH_DEFINED_LAYOUT);
  const result1 = await mdx(MDX_WITH_DEFINED_LAYOUT, {
    mdPlugins: [[plugin, { layout: '../components/Layout1' }]],
  });

  expect(result0).toEqual(result1);
  expect(result1).toMatchSnapshot();
});

it('should not override imported layout', async () => {
  const result0 = await mdx(MDX_WITH_IMPORTED_LAYOUT);
  const result1 = await mdx(MDX_WITH_IMPORTED_LAYOUT, {
    mdPlugins: [[plugin, { layout: '../components/Layout2' }]],
  });

  expect(result0).toEqual(result1);
  expect(result1).toMatchSnapshot();
});

it('should add layout to simple mdx', async () => {
  const result0 = await mdx(MDX_NO_LAYOUT_PLAIN);
  const result1 = await mdx(MDX_NO_LAYOUT_PLAIN, {
    mdPlugins: [[plugin, { layout: '../components/Layout3' }]],
  });

  expect(result0).not.toEqual(result1);
  expect(result1).toMatchSnapshot();
});

it('should add layout to mdx with imports', async () => {
  const result0 = await mdx(MDX_NO_LAYOUT_WITH_IMPORTS);
  const result1 = await mdx(MDX_NO_LAYOUT_WITH_IMPORTS, {
    mdPlugins: [[plugin, { layout: '../components/Layout4' }]],
  });

  expect(result0).not.toEqual(result1);
  expect(result1).toMatchSnapshot();
});
