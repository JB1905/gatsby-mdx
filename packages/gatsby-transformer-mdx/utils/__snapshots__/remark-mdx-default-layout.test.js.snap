// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add layout to mdx with imports 1`] = `
"import Layout from \\"../components\\";
import ___DefaultLayout from \\"../components/Layout4\\";

export default class MDXContent extends React.Component {
  constructor(props) {
    super(props)
    this.layout = ___DefaultLayout
  }
  render() {
    const { components, ...props } = this.props

    return <MDXTag
             name=\\"wrapper\\"
             Layout={this.layout} layoutProps={props}
             components={components}>
<MDXTag name=\\"h1\\" components={components}>{\`Hello world\`}</MDXTag>
<MDXTag name=\\"ul\\" components={components}>
<MDXTag name=\\"li\\" components={components} parentName=\\"ul\\">{\`item 1\`}</MDXTag>
<MDXTag name=\\"li\\" components={components} parentName=\\"ul\\">{\`item 2\`}</MDXTag>
</MDXTag>
<Layout title=\\"Yep\\" />


           </MDXTag>
  }
}"
`;

exports[`should add layout to simple mdx 1`] = `
"import ___DefaultLayout from \\"../components/Layout3\\";

export default class MDXContent extends React.Component {
  constructor(props) {
    super(props)
    this.layout = ___DefaultLayout
  }
  render() {
    const { components, ...props } = this.props

    return <MDXTag
             name=\\"wrapper\\"
             Layout={this.layout} layoutProps={props}
             components={components}><MDXTag name=\\"h1\\" components={components}>{\`Hello world\`}</MDXTag>
<MDXTag name=\\"ul\\" components={components}>
<MDXTag name=\\"li\\" components={components} parentName=\\"ul\\">{\`item 1\`}</MDXTag>
<MDXTag name=\\"li\\" components={components} parentName=\\"ul\\">{\`item 2\`}</MDXTag>
</MDXTag>


           </MDXTag>
  }
}"
`;

exports[`should not override defined layout 1`] = `
"

export default class MDXContent extends React.Component {
  constructor(props) {
    super(props)
    this.layout = ({children}) => <div style={{border: '1px solid red'}}>{children}</div>
  }
  render() {
    const { components, ...props } = this.props

    return <MDXTag
             name=\\"wrapper\\"
             Layout={this.layout} layoutProps={props}
             components={components}>
<MDXTag name=\\"h1\\" components={components}>{\`The Title\`}</MDXTag>
<MDXTag name=\\"p\\" components={components}>{\`Lore ipsum\`}</MDXTag>
           </MDXTag>
  }
}"
`;

exports[`should not override imported layout 1`] = `
"import Layout from \\"somewhere\\";

export default class MDXContent extends React.Component {
  constructor(props) {
    super(props)
    this.layout = Layout
  }
  render() {
    const { components, ...props } = this.props

    return <MDXTag
             name=\\"wrapper\\"
             Layout={this.layout} layoutProps={props}
             components={components}>

<MDXTag name=\\"h2\\" components={components}>{\`Help\`}</MDXTag>
<MDXTag name=\\"blockquote\\" components={components}>
<MDXTag name=\\"p\\" components={components} parentName=\\"blockquote\\">{\`node fix-stuff.js\`}</MDXTag>
</MDXTag>
           </MDXTag>
  }
}"
`;
